### PR TITLE
Fix: addMember page: firebase user and automatic owner addition 

### LINKED ISSUE #250: Frontend: refactor AddMember to use firebase user and detect board owner

### PR DESCRIPTION
**Pull request by shpomp**
# Description

AddMembers page now hides the user from UI when adding members, but is designed in a way to add it automatically as the 'owner' type of member to the board.

Fixes #250 

npm run dev and check http://localhost:3000/boards/new-board - add member

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
- [x] I have commented my code, particularly in hard-to-understand areas, if you code was simple enough mark the box anyway
- [x] I have made corresponding changes to the documentation, if you code was simple enough mark the box anyway
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Comment by senner007** [comment]
@shpomp 
How is the owner added here?

File: src/client/containers/CreateBoardPage/CreateBoardPage.container.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import React, { useState } from 'react';
 import AddMembers from '../AddMembersPage/AddMembersPage.container';
+import { useUser } from '../../firebase/UserContext';
 
 export default function CreateBoard() {
-  const [members, setMembers] = useState([]);
+  const { user } = useUser();
+  const userId = user[0].id;
+  const [members, setMembers] = useState([userId]);
```
Comment (senner007) [inline src/client/containers/CreateBoardPage/CreateBoardPage.container.js:8]: Ok, I see you add the userId to the members before you pass the members list.

**Review by senner007 — APPROVED** [review]

**Comment by shpomp** [comment]
> @shpomp
> How is the owner added here?

The owner is used as the initial state of the members array, that Diny uses to loop through with the addMember endpoint:
```javascript
// ...
import { useUser } from '../../firebase/UserContext';

export default function CreateBoard() {
  const { user } = useUser();
  const userId = user[0].id;
  const [members, setMembers] = useState([userId]);
  const addMember = (id) => {
    setMembers([...members, id]);
  };
// ....
```
The owner (user) is also hidden from the UI now, which is more logical: 

```javascript
// ...
const renderUser = (user) => {
    if (user.id !== userId) {
// ....
```

So the members array is now always starting with members[0] = owner, where endpoint takes care of this, as userId === memberId

**Review by dinythomas89 — APPROVED** [review]
Working fine Karolina

File: src/client/containers/CreateBoardPage/CreateBoardPage.container.js (hunk 1 above)
Comment (senner007) [reply src/client/containers/CreateBoardPage/CreateBoardPage.container.js:8]: It's a bit imperative. Firstly I would call it membersWithOwner
Also, do try and create subcomponents and avoid weaving ternary statements inbetween the jsx. In AddMembers you should have a MembersList component.

**Pull request merged**